rustfmt-core/rustfmt-lib/tests/target/issue-4159.rs

```diff
@@ -0,0 +1,20 @@
+extern "C" {
+    type A: Ord;
+
+    type A<'a>
+    where
+        'a: 'static,
+    ;
```

Comment:
I put the semicolon on a newline for consistency with how other where clauses are formatted. This may be a little awkward, but this code is only valid syntactically, so it is unlikely it will be written often.

Reply:
I agree that this is rather an uncommon way of writing extern type alias. That said, this doesn't look great; we likely fix this in the future so that the trailing semicolon comes next to the last where clause bound. I would like to fix this now if we can.

Could you update the code so that the semicolon will be put on the same line as the last where clause bound?

Reply:
Should we elide the trailing comma?

Reply:
Yes, it should be like the following:

```rust
extern "C" {
    type A<'a>
    where
        'a: 'static;
}
```

Reply:
Cool, I’ll update this later today.